### PR TITLE
feat: Split core and puppeteer features

### LINKED ISSUE #152: puppeteer imported but not in dependencies
## Description

`puppeteer` is [imported](https://github.com/Xetera/ghost-cursor/blob/v1.3.0/src/spoof.ts#L1) into the source code, but it's not into the published [dependencies](https://github.com/Xetera/ghost-cursor/blob/v1.3.0/package.json#L24-L28).

I use `ghost-cursor`, but I don't have `puppeteer` (I use `playwright`).

I think `ghost-cursor` should [export two files](https://nodejs.org/api/packages.html#subpath-exports):

- One with basic functionality that doesn't use `puppeteer`: `path()`
- Another with all features, including those requiring `puppeteer`: `path()`, `createCursor()`...

## To reproduce

- `package.json`

  ```json
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "ghost-cursor": "1.3.0",
      "typescript": "5.7.3"
    }
  }
  ```

- `index.ts`

  ```typescript
  import { path } from "ghost-cursor";
  ```

1. `npm install`
2. `npx tsc index.ts`
   ```
   node_modules/ghost-cursor/lib/mouse-helper.d.ts:1:27 - error TS2307: Cannot find module 'puppeteer' or its corresponding type declarations.

   1 import type { Page } from 'puppeteer';
                            ~~~~~~~~~~~

   node_modules/ghost-cursor/lib/spoof.d.ts:1:50 - error TS2307: Cannot find module 'puppeteer' or its corresponding type declarations.

   1 import { ElementHandle, Page, BoundingBox } from 'puppeteer';
                                                   ~~~~~~~~~~~


   Found 2 errors in 2 files.

   Errors  Files
        1  node_modules/ghost-cursor/lib/mouse-helper.d.ts:1
        1  node_modules/ghost-cursor/lib/spoof.d.ts:1
   ```

### PR DESCRIPTION
**Pull request by regseb**
Fix https://github.com/Xetera/ghost-cursor/issues/152

Split features into two files:

- `core` for puppeteer independent features : `path()`
- `spoof` (default) for puppeteer related features and also `core`: `createCursor()`, `path()`

### CODE REVIEW

**Comment by regseb** [comment]
@Niek Now that the refactor has been merged, can you review this pull request?

File: package.json [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
   "repository": "https://github.com/Xetera/ghost-cursor",
+  "exports": {
+    ".": {
+      "types": "./lib/spoof.d.ts",
+      "default": "./lib/spoof.js"
+    },
+    "./core": {
+      "types": "./lib/core.d.ts",
+      "default": "./lib/core.js"
+    },
```
[the remark this answers is not in the document]
Comment (Niek) [reply package.json:14]: @hat do you think @regseb?

[the remark this answers is not in the document]
Comment (Niek) [reply package.json:14]: I don't think people imported these directly, since they are not documented. Right, @regseb?

File: src/core.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  const length = curve.length() * 0.8
+
+  const speed = optionsResolved.moveSpeed !== undefined && optionsResolved.moveSpeed > 0
+    ? (25 / optionsResolved.moveSpeed)
+    : Math.random()
+  const baseTime = speed * MIN_STEPS
+  const steps = Math.ceil((Math.log2(fitts(length, width) + 1) + baseTime) * 3)
+  const re = curve.getLUT(steps)
+  return clampPositive(re, optionsResolved)
+}
```
[the remark this answers is not in the document]
Comment (Niek) [reply src/core.ts:66]: @copilot create a unit test based on [this feedback](https://github.com/Xetera/ghost-cursor/pull/171#discussion_r3088919076). make sure to test the most common edge-cases.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (regseb) [reply package.json:14]: It depends on the package's main purpose.

1. Humanize mouse movements in Puppeteer (and secondary generate human-like movement).

   ```json
   {
     ".": "./lib/spoof.js",
     "./core": "./lib/core.js"
   }
   ```

2. Generate human-like movement (and secondary provide integration for Puppeteer).

   ```json
   {
     ".": "./lib/core.js",
     "./puppeteer": "./lib/spoof.js"
   }
   ```

With the second solution: there is a breaking change, but it’s easier to add a new integration.

[the remark this answers is not in the document]
Comment (regseb) [reply package.json:14]: Yes, it's unlikely that anyone would import a deeply nested file.

- It's not common in npm to import deeply nested files.
- The existence of these files isn't documented.
- I don't see what functionality a user would want to access in these files.

[the remark this answers is not in the document]
Comment (regseb) [reply package.json:14]: @Niek Should I modify the exports (and release a breaking change)? Or should I keep the exports from the PR, which maintains backward compatibility?
